juntagrico/util/sessions.py

```diff
@@ -0,0 +1,122 @@
+from juntagrico.config import Config
+
+
+class SessionObjectManager:
+    def __init__(self, request, key, data_type):
+        self._request = request
+        self._key = key
+        if key in request.session:
+            self.data = request.session.get(key)
+        else:
+            self.data = data_type()
+            request.session[self._key] = self.data
+
+    def store(self):
+        # only store if key exists, otherwise session was flushed
+        if self._key in self._request.session and not self.data.cleared:
+            self._request.session[self._key] = self.data
+
+
+class SessionObject:
+    def __init__(self):
+        self.cleared = False
+
+    def clear(self):
+        self.cleared = True
+        self.__init__()
+
+
+class CSSessionObject(SessionObject):
+    def __init__(self):
+        super().__init__()
+        self._main_member = None
+        self._co_members = []
+        self.co_members_done = False
+        self.subscriptions = {}
+        self.depot = None
+        self.start_date = None
+        self.edit = False
+
+    @property
+    def main_member(self):
+        return self._main_member
+
+    @main_member.setter
+    def main_member(self, new_main_member):
+        # transfer previously selected shares
+        new_main_member.new_shares = getattr(self._main_member, 'new_shares', None)
+        self._main_member = new_main_member
+
+    @property
+    def co_members(self):
```

Comment:
why not use 
return self._co_members.copy()

Reply:
get_co_member used to do something more at one point. Will fix this.